analyzer/src/main/kotlin/curation/FilePackageCurationProvider.kt

```diff
@@ -28,11 +28,13 @@ import org.ossreviewtoolkit.model.PackageCuration
 import org.ossreviewtoolkit.model.readValueOrDefault
 
 /**
- * A [PackageCurationProvider] that loads [PackageCuration]s from a single file. Supports all file formats specified
- * in [FileFormat].
+ * A [PackageCurationProvider] that loads [PackageCuration]s from all [curationFiles]. Supports all file formats
+ * specified in [FileFormat].
  */
-class FilePackageCurationProvider(curationFile: File) : PackageCurationProvider {
-    internal val packageCurations by lazy { curationFile.readValueOrDefault(emptyList<PackageCuration>()) }
+class FilePackageCurationProvider(curationFiles: List<File>) : PackageCurationProvider {
+    internal val packageCurations by lazy {
+        curationFiles.map { it.readValueOrDefault(emptyList<PackageCuration>()) }.flatten()
```

Comment:
Please put reading a single file into `try / catch` and log a message when reading fails. 
As-is it's hard to figure out for which of the files the deserialization failed.
Also see the `PackageConfigurationProvider` class.

Reply:
Speaking about the `(Simple)PackageConfigurationProvider)`, I plan to refactor the class a bit to *also* allow multiple package configurations in a single file when multiple files are spread across a directory tree. And we should probably then do the same here to align the behavior. Any objections @fviernau?

Reply:
> "I plan to refactor the class a bit to also allow multiple package configurations in a single file"

Hm, not sure what exactly you have in mind. I guess that would change the file format of these package configurations, is that correct? What's the reason to do that?

Regarding this commit vs. `(Simple)PackageConfigurationProvider`, I find it quite nice that is has `configurations: Collection<PackageConfiguration>` as opposed to `List<File>`. I believe this PR can consider aligning with that approach.

Reply:
> I guess that would change the file format of these package configurations, is that correct?

No, it would simply be tried to either read `PackageConfiguration` or `List<PackageConfiguration>` from files found by `findFilesWithKnownExtensions`.

> What's the reason to do that?

That way you could have a directory hierarchy that nests package type / namespace / name, but put different `PackageConfiguration`s for different versions into the same file. But the current mode of one `PackageConfiguration` per file in the tree would also still work.

Also, @MarcelBochtler's change here goes into the direction of always reading `List<PackageCuration>` from curation files. I'd find it confusing for the user to remember that, in case of a directory hierarchy, for `PackageConfiguration`s you must put only one per file, but for `PackageCuration` you must put multiple per file. IMO we should align that by allow both formats for both file types.

Reply:
I find one package configuration quite aligned with the `ort.yml` so I'm not with you in the confusing user bit. However, I agree with your argumentation. So I'm fine implementing it as you just outlined.

Reply:
> Please put reading a single file into try / catch and log a message when reading fails.
As-is it's hard to figure out for which of the files the deserialization failed.
Also see the PackageConfigurationProvider class.

Good point.
I opted to catch `Exception` instead of `IOException` which is done in `PackageConfigurationProvider` to also catch e.g. `IllegalArgumentExceptions` if the file is not valid.